src/lib/paymentOptions/NimiqPaymentOptions.ts

```diff
@@ -0,0 +1,186 @@
+import { TX_VALIDITY_WINDOW, TX_MIN_VALIDITY_DURATION } from '../Constants';
+import { Currency, PaymentMethod, PaymentOptions } from '../PublicRequestTypes';
+import { ParsedPaymentOptions } from '../RequestTypes';
+import { toNonScientificNumberString } from '@nimiq/utils';
+
+export interface NimiqProtocolSpecific {
+    fee?: number | string;
+    feePerByte?: number | string;
+    validityDuration?: number;
+    flags?: number;
+    sender?: string;
+    forceSender?: boolean;
+    recipient?: string;
+    recipientType?: Nimiq.Account.Type;
+}
+
+export type ParsedNimiqProtocolSpecific = Omit<NimiqProtocolSpecific, 'sender' | 'recipient' | 'fee' | 'feePerByte'>
+    & Required<Pick<NimiqProtocolSpecific, 'forceSender' | 'flags' | 'validityDuration'>> & {
+    sender?: Nimiq.Address,
+    recipient?: Nimiq.Address,
+    fee: number,
+    feePerByte: number,
+};
+
+export type NimiqDirectPaymentOptions = PaymentOptions<Currency.NIM, PaymentMethod.DIRECT>;
+
+export class ParsedNimiqDirectPaymentOptions extends ParsedPaymentOptions<Currency.NIM, PaymentMethod.DIRECT> {
+    public amount: number;
+    private extraData: Uint8Array;
+
+    public constructor(options: NimiqDirectPaymentOptions, extraData: Uint8Array) {
+        super(options);
+        this.extraData = extraData;
+        this.amount = Number.parseInt(toNonScientificNumberString(options.amount), 10);
+
+        let sender: Nimiq.Address | undefined;
+        if (options.protocolSpecific.sender !== undefined) {
+            try {
+                sender = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.sender);
+            } catch (err) {
+                throw new Error('If provided, sender must be a valid user friendly address string');
+            }
+        }
+
+        let recipient: Nimiq.Address | undefined;
+        if (options.protocolSpecific.recipient !== undefined) {
+            try {
+                recipient = Nimiq.Address.fromUserFriendlyAddress(options.protocolSpecific.recipient);
```

Comment:
We could use `Nimiq.Address.fromString()` here to also allow HEX and base64 encodings.

Reply:
I do not see any occurrence of `Nimiq.Address.fromString` in the entirety of the Hub. I'd say if we want to support that we should do it in a different PR.

Reply:
Since you created an issue for it I will resolve this here.